src/ctapipe/monitoring/calculator.py

```diff
@@ -0,0 +1,327 @@
+            )
+            # Slice the dl1 table according to the previously calculated start and end.
+            table_sliced = table[slice_start:slice_end]
+            # Run the stats aggregator on the sliced dl1 table with a chunk_shift
+            # to sample the period of trouble (carflashes etc.) as effectively as possible.
+            # Checking for the length of the sliced table to be greater than the ``chunk_size``
+            # since it can be smaller if the last two chunks are faulty. Note: The two last chunks
+            # can be overlapping during the first pass, so we simply ignore them if there are faulty.
+            if len(table_sliced) > aggregator.chunk_size:
+                aggregated_stats_secondpass.append(
+                    aggregator(
+                        table=table_sliced,
+                        masked_pixels_of_sample=masked_pixels_of_sample,
+                        col_name=col_name,
+                        chunk_shift=self.chunk_shift,
+                    )
+                )
+        # Stack the aggregated statistics of each faulty chunk
+        aggregated_stats_secondpass = vstack(aggregated_stats_secondpass)
+        # Detect faulty pixels with multiple instances of OutlierDetector of the second pass
+        outlier_mask_secondpass = np.zeros_like(
+            aggregated_stats_secondpass["mean"], dtype=bool
+        )
+        for (
+            aggregated_val,
+            outlier_detector,
+        ) in self.outlier_detectors.items():
+            outlier_mask_secondpass = np.logical_or(
+                outlier_mask_secondpass,
+                outlier_detector(aggregated_stats_secondpass[aggregated_val]),
+            )
+        # Add the outlier mask to the aggregated statistics
+        aggregated_stats_secondpass["outlier_mask"] = outlier_mask_secondpass
+        aggregated_stats_secondpass["is_valid"] = self._get_valid_chunks(
+            outlier_mask_secondpass
+        )
+        return aggregated_stats_secondpass
+
+    def _get_valid_chunks(self, outlier_mask):
+        """
+        Identify valid chunks based on the outlier mask.
+
+        This method processes the outlier mask to determine which chunks of data
+        are considered valid or faulty. A chunk is marked as faulty if the percentage
+        of outlier pixels exceeds a predefined threshold ``faulty_pixels_threshold``.
+
+        Parameters
+        ----------
+        outlier_mask : numpy.ndarray
+            Boolean array indicating outlier pixels. The shape of the array should
+            match the shape of the aggregated statistics.
+
+        Returns
+        -------
+        numpy.ndarray
+            Boolean array where each element indicates whether the corresponding
+            chunk is valid (True) or faulty (False).
+        """
+        # Check if the camera has two gain channels
+        if outlier_mask.shape[1] == 2:
+            # Combine the outlier mask of both gain channels
```

Comment:
Why? The two gains can be independent

Reply:
This function only detects whether the chunk is faulty or valid. Therefore, we need to calculate the fraction of faulty pixels in the camera. Here I define a pixel as faulty if one of the gains are faulty. Mind that definition is only used to decide if a chunk is faulty in order to calculate the second pass around that chunk. The outlier mask is stored for two gains, so one would have the information when _camcalib coefficients_ are applied.

Reply:
Wouldn't it make sense to process both gains independently?

Reply:
That would complicate things but in principle that would be possible. However, the goal of the functionality is to detect time periods of troubles, mainly car flashes. Therefore, both gains should be highly correlated anyway.